From e2817aedf32cba8bc4f8aae4debb1f41b6573b8f Mon Sep 17 00:00:00 2001
From: Venkateshwar Rao Gannavarapu <venkateshwar.rao.gannavarapu@xilinx.com>
Date: Thu, 23 Jan 2020 17:47:42 +0530
Subject: [PATCH 0964/1852] dt-bindings: Add device tree binding for xilinx
 mpeg2tsmux driver

commit 24672d5caae54409330be966a9d4e237d77b5bdf from
https://github.com/Xilinx/linux-xlnx.git

This patch adds the device tree bindings for xilinx mpeg transport muxer

Signed-off-by: Venkateshwar Rao Gannavarapu <venkateshwar.rao.gannavarapu@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/staging/xlnx_tsmux/dt-binding.txt | 28 +++++++++++++++++++++++
 1 file changed, 28 insertions(+)
 create mode 100644 drivers/staging/xlnx_tsmux/dt-binding.txt

diff --git a/drivers/staging/xlnx_tsmux/dt-binding.txt b/drivers/staging/xlnx_tsmux/dt-binding.txt
new file mode 100644
index 000000000000..e4a7095d92e1
--- /dev/null
+++ b/drivers/staging/xlnx_tsmux/dt-binding.txt
@@ -0,0 +1,28 @@
+The Xilinx mpegtsmux IP reads the elementary streams from memory and
+writes the MPEG2 TS(transport stream) to memory.
+
+The mpeg2 ts muxer follows the dma descriptor based approach. Each DMA
+descriptor contains information about each of the elementary stream
+buffer properties and buffer address. It reads the descriptors one after
+the other and generates the TS packets with the information in the
+descriptor. The IP writes the generated TS packets at the output buffer
+address.
+
+Required properties:
+
+- compatible: must be "xlnx,tsmux-1.0"
+- interrupts: interrupt number
+- interrupts-parent: phandle for interrupt controller
+- reg: base address and size of the IP core
+- clock-names: must contain "ap_clk"
+- clocks: phandle to AXI Lite
+
+Example:
+	ts2mux: ts2mux@0xa0200000 {
+		compatible = "xlnx,tsmux-1.0";
+		interrupt-parent = <&gic>;
+		interrupts = <0 90 4>;
+		reg = <0x0 0xa0200000 0x0 0x30000>;
+		clock-names = "ap_clk";
+		clocks = <&misc_clk_0>;
+	};
-- 
2.31.1

